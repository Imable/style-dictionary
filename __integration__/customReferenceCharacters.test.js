const fs = require('fs-extra');
const StyleDictionary = require('../index');
const {buildPath} = require('./_constants');

describe('integration', () => {
  describe('custom reference characters', () => {
    StyleDictionary.extend({
      tokens: {
        color: {
          red: { value: '#f00' },
          error: { value: '🐈color/red/value🐈‍⬛' }
        },
        // this should NOT resolve
        test: { value: '{color.red.value}' }
      },
      separator: '/',
      opening_character: '🐈',
      closing_character: '🐈‍⬛',
      platforms: {
        css: {
          transformGroup: 'css',
          buildPath,
          files: [{
            destination: 'variables.css',
            format: 'css/variables'
          }]
        }
      }
    }).buildAllPlatforms();
    const output = fs.readFileSync(`${buildPath}variables.css`, {encoding:'UTF-8'});
    it(`should match snapshot`, () => {
      expect(output).toMatchSnapshot();
    });
  });
});