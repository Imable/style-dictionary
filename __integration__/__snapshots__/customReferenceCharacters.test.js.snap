// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration custom reference characters should match snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

:root {
  --color-red: #ff0000;
  --color-error: #ff0000;
}
"
`;
